code/modules/mapfluff/ruins/spaceruin_code/hauntedtradingpost.dm

```diff
@@ -194,19 +199,9 @@ GLOBAL_LIST_EMPTY(tripwire_suicide_pact)
 	INVOKE_ASYNC(src, TYPE_PROC_REF(/atom, interact), victim)
 	if(multiuse && uses_remaining != 1)
 		return
-	if(suicide_pact&& suicide_pact_id)
-		for (var/obj/machinery/button/door/invisible_tripwire/pact_member in GLOB.tripwire_suicide_pact)
-			if(src.suicide_pact_id == pact_member.suicide_pact_id)
-				qdel(pact_member)
-	qdel(src)
-
-
-/obj/machinery/button/door/invisible_tripwire/Destroy()
-	if(donk_ai_slave)
-		GLOB.selfdestructs_when_boss_dies -= src
 	if(suicide_pact && suicide_pact_id)
-		GLOB.tripwire_suicide_pact -= src
-	return ..()
+		SEND_SIGNAL(src, COMSIG_PUZZLE_COMPLETED)
```

Comment:
Make a new comsig for this, please, otherwise it's good.

Reply:
this signal can be listened to by other objects in the queue such as the indestructible puzzle fields so im not so sure about that

Reply:
Well then, can't we place this signal in a separate proc at least? That way we will have a single source of the signal, even if called by different entities.

Reply:
that means the tripwire can trigger other puzzle objects _intentionally_

Reply:
No, it's just going to be a container for the signal like:
```
/atom/movable/proc/complete_puzzle()
	SEND_SIGNAL(src, COMSIG_PUZZLE_COMPLETED)
```

Reply:
what would that even change

Reply:
Signals are generally meant to be sent in one and one place only. Usually, if that's not the case, something's wrong.  We need to keep some sanity around the datum component system, otherwise we'd end up with structures that send human-related signals for snowflakey and arcane reasons